src/apps/Altinn.AccessManagement/src/Altinn.AccessManagement.Api.Internal/Controllers/Bff/ConsentController.cs

```diff
@@ -0,0 +1,158 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Altinn.AccessManagement.Api.Internal.Controllers.Bff
+{
+    /// <summary>
+    /// API controller for managing consent information for end users.
+    /// All endpoints are accessible only from the Altinn Portal to ensure that end users are properly informed about the details of their consents.
+    /// The controller enforces the portal scope for authorization to access its methods.
+    /// </summary>
+    [Route("accessmanagement/api/v1/bff")]
+    [ApiController]
+    [Authorize(Policy = AuthzConstants.SCOPE_PORTAL_ENDUSER)]
+    public class ConsentController(IConsent consentService) : ControllerBase
+    {
+        private readonly IConsent _consentService = consentService;
+
+        /// <summary>
+        /// Get a specific consent. 
+        /// Requires the following.
+        /// User is authenticated
+        /// User has write access to access management for the party that is requested to consent (from party)
+        /// User is authorized to delegate the rights that are requested. Either by having the right themselves or being the main administrator
+        /// </summary>
+        [HttpGet]
+        [Route("consentrequests/{requestId}", Name ="bffgetconsentrequest")]
+        public async Task<IActionResult> GetConsentRequest([FromRoute] Guid requestId, CancellationToken cancellationToken = default)
+        {
+            Guid? performedBy = UserUtil.GetUserUuid(User);
+            if (performedBy == null)
+            {
+                return Unauthorized();
+            }
+
+            ConsentPartyUrn performedByParty = ConsentPartyUrn.PartyUuid.Create(performedBy.Value);
+
+            Result<ConsentRequestDetails> consentRequest = await _consentService.GetRequest(requestId, performedByParty, true, cancellationToken);
+            if (consentRequest.IsProblem)
+            {
+                return consentRequest.Problem.ToActionResult();
+            }
+
+            return Ok(consentRequest.Value.ToConsentRequestDetailsBFF());
+        }
+
+        /// <summary>
+        /// Get a specific consent. 
+        /// Requires the following.
+        /// User is authenticated
+        /// User is have write access to access management for the party that is requestesd to consent (from party)
+        /// User is authorized to delegated the rights that is requested. Either by having the right self or beeing the main administrator
+        /// </summary>
+        [HttpGet]
+        [Route("consents/{requestId}/")]
+        public async Task<IActionResult> GetConsent([FromRoute] Guid requestId, CancellationToken cancellationToken = default)
+        {
+            Guid? performedBy = UserUtil.GetUserUuid(User);
+            if (performedBy == null)
+            {
+                return Unauthorized();
+            }
+
+            Result<Consent> consent = await consentService.GetConsent(requestId, cancellationToken);
```

Comment:
Hvor skjer autorisasjon av at performedBy er autorisert for å hente Samtykket?

Reply:
Lagt til autorisasjon i controller